### ISSUE TITLE
v3.1.7 triggering fatal error

### ISSUE THREAD

**Issue by sukafia** [woocommerce/facebook-for-woocommerce#2679]
## ✍️ Describe the bug

The latest version (3.1.7) of Facebook for WooCommerce is causing fatal errors. 

## 🚶‍♀️ Steps to reproduce

1. Update to the latest version
2. Visit the WooCommerce order page (WooCommerce > Orders > Open an order)
3. Notice a fatal error is triggered 

![2rhYPc.png](https://github.com/woocommerce/facebook-for-woocommerce/assets/60121732/419215d3-af40-4da6-b348-42786a8153a1)

## 🗃 Error

```
An error of type E_ERROR was caused in line 1643 of the file /home/cus0/public_html/wp-content/plugins/facebook-for-woocommerce/facebook-commerce.php. Error message: Uncaught Error: Call to a member function is_type() on bool in /home/cus222mizer/public_html/wp-content/plugins/facebook-for-woocommerce/facebook-commerce.php:1643
Stack trace:
#0 /home/cus0/public_html/wp-includes/class-wp-hook.php(324): WC_Facebookcommerce_Integration->display_batch_api_completed('shop_order', Object(WP_Post))
#1 /home/cus0/public_html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#2 /home/cus0/public_html/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#3 /home/cus0/public_html/wp-admin/includes/meta-boxes.php(1714): do_action('add_meta_boxes', 'shop_order', Object(WP_Post))
#4 /home/cus20/public_html/wp-admin/edit-form-advanced.php(271): register_and_do_post_meta_boxes(Object(WP_Post))
#5 /home/cus20/public_html/wp-admin/post.php(206): require('/home/cus222miz...')
#6 {main}
  thrown
```

**Comment by sukafia** [comment]
Affected users:

https://wordpress.org/support/topic/3-1-7-breaks-woocommerce-order-pages/
https://wordpress.org/support/topic/new-update-giving-errors/

**Comment by davidhefti** [comment]
same issue after Plugin update

Ein Fehler vom Typ E_ERROR wurde in der Zeile 1643 der Datei /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-content/plugins/facebook-for-woocommerce/facebook-commerce.php verursacht. Fehlermeldung: Uncaught Error: Call to a member function is_type() on bool in /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-content/plugins/facebook-for-woocommerce/facebook-commerce.php:1643
Stack trace:
#0 /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-includes/class-wp-hook.php(324): WC_Facebookcommerce_Integration->display_batch_api_completed()
#1 /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters()
#2 /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-includes/plugin.php(517): WP_Hook->do_action()
#3 /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-admin/includes/meta-boxes.php(1714): do_action()
#4 /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-admin/edit-form-advanced.php(271): register_and_do_post_meta_boxes()
#5 /home/879604.cloudwaysapps.com/bbjatrzruk/public_html/wp-admin/post.php(206): require('/home/879604.cl...')
#6 {main}
  thrown

**Comment by gabriel-fuentes** [comment]
Another report here: 7527475

**Comment by carolframen** [comment]
Another report: 7527462

**Comment by sukafia** [comment]
https://wordpress.org/support/topic/new-update-breaks-order-page-wc-hpos/
https://wordpress.org/support/topic/fatal-error-uncaught-error-call-to-a-member-function-is_type-on-bool/

**Comment by carolframen** [comment]
https://wordpress.org/support/topic/new-update-is-giving-errors/

**Comment by yzmp** [comment]
I got the exact same error after upgrading to 3.1.7

`2024-01-03T12:37:50+00:00 CRITICAL Uncaught Error: Call to a member function is_type() on bool in /facebook-for-woocommerce/facebook-commerce.php:1643`

**Comment by ibndawood** [comment]
We've fixed this issue in `v3.1.8`. Please update the plugin to resolve this issue. Thank you.
